### DISCUSSION TITLE
Email input component

### DISCUSSION THREAD

**Discussion by NemirStephan** [Legal-and-General/canopy#266] [Ideas] [answered]
I'd like to propose a new email input component that will automatically validate the email (structure, not whether it is an active email or not). And have a build in validator. There could be a default value for the validation message that can be overridden

**Comment by andy-polhill** [comment, marked as answer]
@NemirStephan do we know if it looks visually different from the standard input? I'm just wondering if it needs a special component, or whether we can just use the standard input with an Angular email validator?

**Reply by NemirStephan** [reply]
I think its pretty much the same as a standard input, so yes, probably just needs a standard angular email validator, but I was thinking that we could have a default validation message for consistency that can also be overridden so it's not too constrained.

**Reply by alexcanning** [reply]
Definitely should be a pattern, so we consistently ask users for emails in the same way and as Nemir suggests, provide error messages the same way.

Pattern name would be:
"Ask user for email address"

**Reply by NemirStephan** [reply]
We have agreed that for now at least there is no need for a separate component. We should however add a new 'For email' example to storybook under the Form/Input component. Developers should be able to copy and paste the example giving us consistency. For example, something like this. @alexcanning, would you be able to confirm the copy for the error message?

mycomponent.component.html
```html
<form
  [formGroup]="emailForm"
  #form="ngForm">
  <lg-input-field>
    Email address
    <input
      lgInput
      type="email"
      formControlName="email"
      size="30"
    />
    <lg-validation *ngIf="isControlInvalid(email, form) && email.hasError('email')">
      Your email address is not valid
    </lg-validation>
  </lg-input-field>
</form>
```

mycomponent.component.ts
```typescript
import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
import { LgErrorStateMatcher } from '@legal-and-general/canopy';


@Component({
  selector: 'app-mycomponent',
  templateUrl: './mycomponent.component.html'
})
export class MyComponent {
  emailForm: FormGroup = this.fb.group({
    email: ['', Validators.email],
  });

  constructor(private fb: FormBuilder, private errorState: LgErrorStateMatcher) { }

  get email(): AbstractControl {
    return this.emailForm.controls.email;
  }

  public isControlInvalid(control, form) {
    return this.errorState.isControlInvalid(control, form);
  }
}

```
